### DISCUSSION TITLE
resetting shortcuts in private.ini doesn't work correctly for more than one submenu

### DISCUSSION THREAD

**Discussion by u-fischer** [WinEdt-Team/WinEdt-Support#49] [1.2 Bugs and Issues]
I updated yesterday to 11.2 and then had problems with the activation of some settings in my Private.ini. It mostly contains changes in the shortcuts: I disable various shortcuts that I don't need and change various other. 

The general setup is 

~~~~
[MAIN_MENU**]

MENU="&Document"
   ITEM="LOG_File"
   SHORTCUT="16460::Ctrl+L"
   ITEM="STD_Output_File"
   SHORTCUT=""
END="&Document"   
~~~~

After the update some of the shortcut redefinition failed and after some digging and testing I found out that it stops working if there are two sections with submenus.  So if I do something like the following, the shortcut for BibTeX is adapted but the follow up change for LaTeX is ignored.

~~~~
MENU="TeX_Menu"
  SUBMENU="BibIndex>"
     ITEM="BibTeX"  
     SHORTCUT=""  
 END="BibIndex>"  
 SUBMENU="TeXExtra>"
     ITEM="LaTeX"
     SHORTCUT="24652::Shift+Ctrl+L"  
 END="TeXExtra>"  
END="TeX_Menu"   
~~~~

It started to work again after I splitted the instructions:

~~~~
MENU="TeX_Menu"
  SUBMENU="BibIndex>"
     ITEM="BibTeX"  
     SHORTCUT=""  
 END="BibIndex>"  
END="TeX_Menu"   

MENU="TeX_Menu"
 SUBMENU="TeXExtra>"
     ITEM="LaTeX"
     SHORTCUT="24652::Shift+Ctrl+L"  
 END="TeXExtra>"  
END="TeX_Menu"   
~~~~

I do not know if this is a new problem: The TeX menu has been rearranged a bit and it is possible that previously I simply didn't notice that some of my redefinitions were ignored.

**Comment by Alex-Simonic** [comment]
This has always been like this but now resulted in an oddity because of the new submenus. Submenus (and their END) should not be specified in the `[MENU**]` sections. In your case, the following will do fine (regardless of the submenus):

```
MENU="TeX_Menu"
     ITEM="BibTeX"  
     SHORTCUT=""  
     ITEM="LaTeX"
     SHORTCUT="24652::Shift+Ctrl+L"  
END="TeX_Menu"   
```

You may want to drop the submenus when changing shortcuts like this. Or am I missing something?

**Comment by u-fischer** [comment]
> Submenus (and their END) should not be specified in the [MENU**] sections.

Hm, I need to recheck that. I think I added that at some time because something didn't work.

> You may want to drop the submenus when changing shortcuts like this. Or am I missing something?

I'm not sure what you mean here. The (sub)menus can stay as they are (but I don't mind either if they change). I rarely use the menu to start something. But all entries should stay *visible*, as I sometimes forget which shortcut calls which engine and then I can check.

**Reply by Alex-Simonic** [reply]
I meant that you don't have to specify the submenu in which an existing ITEM resides to change its definition (as long as items have a unique name ID -- not to be mistaken for the caption). Of course, everything remains visible and organized in the submenus as it was. The idea of the `[MENU**]` section is to allow changing the properties (like Shortcuts) of the existing menu items without modifying the default MainMenu.ini. If for whatever reason you must specify the submenu then you will have to split it one submenu at a time (as you did in your second example that works). But a simpler version as I suggested will do the same in one step.

**Reply by Alex-Simonic** [reply]
The oddities with specifying the `END=" submenu>"` when using the `[MENU**]` section have been there since this feature was introduced. But it should not work like this. I have now fixed it so that your original construction will work properly with (or without) specified submenus.

I will make another build with this fix. Also, WinEdtPDF will now allow custom scale (zoom) specifications for printing (a few users were asking for it).
